test/dhcp/server_isc_test.rb

```diff
@@ -82,8 +82,24 @@ def test_poap_quirks
   def test_loadSubnets_loads_managed_subnets
     subnets = @dhcp.loadSubnets
```

Comment:
I know that this was here before, but it wasn't a very good test: `loadSubnets` returns subnets that were processed, not the data that was stored.

Another issue is that this tests too many things; which will make it harder to pinpoint an issue should code change. Could you break this up to test specific cases, for example handling of `domain_name_server` option when it's not present, when only one server is present, when both are present, etc. 

It is probably easier to work directly with subnet_service (see `@subnet_service` in setup -- you can query it for relevant subnets, as opposed to parsing the hash returned by `@dhcp.subnets`.